src/util/to-cached.ts

```diff
@@ -0,0 +1,130 @@
+/* eslint-disable no-await-in-loop */
+import {ensureArgObject, pickBy} from './index'
+import {Command} from '../command'
+import {Plugin as IPlugin} from '../interfaces/plugin'
+import {aggregateFlags} from './aggregate-flags'
+import {defaultToCached} from './default-to-cached'
+
+export async function toCached(cmd: Command.Class, plugin?: IPlugin, respectNoCacheDefault = false): Promise<Command.Cached> {
+  const flags = {} as {[k: string]: Command.Flag.Cached}
+
+  // In order to collect static properties up the inheritance chain, we need to recursively
+  // access the prototypes until there's nothing left. This allows us to combine baseFlags
+  // and flags as well as add in the json flag if enableJsonFlag is enabled.
+  const mergePrototype = (result: Command.Class, cmd: Command.Class): Command.Class => {
+    const proto = Object.getPrototypeOf(cmd)
+    const filteredProto = pickBy(proto, v => v !== undefined) as Command.Class
+    return Object.keys(proto).length > 0 ? mergePrototype({...filteredProto, ...result} as Command.Class, proto) : result
+  }
+
+  const c = mergePrototype(cmd, cmd)
+
+  const cmdFlags = aggregateFlags(c.flags, c.baseFlags, c.enableJsonFlag)
+
+  for (const [name, flag] of Object.entries(cmdFlags)) {
```

Comment:
this code has always bothered me.  We don't have to do it now, but FYI: 
https://github.com/oclif/core/pull/763/files#diff-615dacd7000fa06d7afef0f22b7b6d9267c6219fbfc9fa0315e51a5f5d963755